### ISSUE TITLE
docs: [@farcaster/js] condense type annotations for function signatures

### ISSUE THREAD

**Issue by varunsrin** [farcasterxyz/hubble#614]
**What is the feature you would like to implement?**

Doc generation should either condense types to the highest level or eliminate them entirely. 

**Why is this feature important?**

It's quite hard to parse docs because of the lengthy type annotations which are also not very useful. 

<img width="619" alt="Screen Shot 2023-02-25 at 5 29 59 PM" src="https://user-images.githubusercontent.com/528882/221387386-a233f335-ac00-4785-904a-475ec08f29f2.png">

**Will the protocol spec need to be updated??**
No

**How should this feature be built?**

The ideal type annotation in this case would be `makeCastAdd(bodyJson, dataOptions, signer): HubAsyncResult<CastAddMessage>`. If that isn't possible to configure, omit the type annotations entirely.

**Comment by varunsrin** [comment]
@vinliao could you take this one?

**Comment by vinliao** [comment]
See: https://github.com/vinliao/hubble/blob/md-docs-hide-sig/packages/js/docs/modules.md#functions. Something like this? This means we have to manually document the params and return types?

**Comment by varunsrin** [comment]
ideally it would autogenerate `makeCastAdd(bodyJson, dataOptions, signer): HubAsyncResult<CastAddMessage>`, but i'm not sure if that's possible with typedoc

**Comment by vinliao** [comment]
There doesn't seem to be native support for that. Current solution (the link above) requires writing a custom plugin just to remove the type signature. See more: https://github.com/TypeStrong/typedoc/issues/1831. Should I manually add the return types to all the functions?

**Comment by vinliao** [comment]
See: https://github.com/vinliao/hubble/blob/md-docs-hide-sig/packages/js/docs/modules.md#functions. Unfortunate it doesn't reduce to its "readable form" (it says things like `HubAsyncResult<Readonly<Object>>` which means nothing). But functions with shallow types actually look pretty OK (`signMessageHash(hash): HubAsyncResult<Uint8Array>`): https://github.com/vinliao/hubble/blob/md-docs-hide-sig/packages/js/docs/classes/Eip712Signer.md#signmessagehash.

**Comment by varunsrin** [comment]
Hmm, then I think we may need to manually clean it up to make it useful to developers. It's kind of annoying to do, so feel free to also add TODO's and we can tackle them piecemeal. 

Also, right now the params and return values are specified twice, once as a one liner and one in a more broken down form in a table. I think we can be brief and just do the latter.

**Comment by vinliao** [comment]
> I think we can be brief and just do the latter.

Be brief and... do the table? Maybe you mean one-liner? Something like this: https://github.com/vinliao/hubble/blob/md-docs-hide-sig/packages/js/docs/modules.md#makecastadd?

**Comment by varunsrin** [comment]
sorry, should have been more precise with my explanation. something like this:

### makeCastAdd

Creates a CastAdd message that can be used to add a Cast to the Farcaster network. 

#### Parameters

| Name | Type | Description |
| :------ | :------ |  :------ |
| `bodyJson` | [`CastAddBody`](modules/types.md#castaddbody) | A valid CastAdd body | 
| `dataOptions` | `MessageDataOptions` | Optional arguments to construct the Cast | 
| `signer` | `Ed25519Signer` | A valid Signer that will sign the message |

#### Return Value

| Name | Type | Description |
| :------ | :------ |  :------ |
| `HubAsyncResult<CastAddMessage>` | [`CastAddBody`](modules/types.md#castaddbody) |  A Result that contains the valid CastAddMessage |



**`Example`**

```typescript
import {
  Client,
  Ed25519Signer,
  makeCastAdd,
  makeCastRemove,
  types,
} from '@farcaster/js';
import * as ed from '@noble/ed25519';

const rpcUrl = '<rpc-url>';
const client = new Client(rpcUrl);

const privateKeyHex = '86be7f6f8dcf18...'; // EdDSA hex private key
const privateKey = ed.utils.hexToBytes(privateKeyHex);

// _unsafeUnwrap() is used here for simplicity, but should be avoided in production
const ed25519Signer = Ed25519Signer.fromPrivateKey(privateKey)._unsafeUnwrap();

const dataOptions = {
  fid: -9999, // must be changed to fid of the custody address, or else it will fail
  network: types.FarcasterNetwork.FARCASTER_NETWORK_DEVNET,
};

const cast = await makeCastAdd({ text: 'hello world' }, dataOptions, ed25519Signer);
await client.submitMessage(cast._unsafeUnwrap());
```

**Comment by varunsrin** [comment]
closing this issue since it has been resolved by our recent rewrite of the documentation
